service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunsPoller.java

```diff
@@ -84,39 +86,53 @@ public List<Run> updateRuns(List<Run> runs) {
 
     try {
       // Filter only updatable runs:
-      List<Run> updatableRuns = runs.stream().filter(r -> r.status().nonTerminal()).toList();
+      Set<Run> updatableRuns =
+          runs.stream().filter(r -> r.status().nonTerminal()).collect(Collectors.toSet());
 
       increaseEventCounter("status update required", updatableRuns.size());
 
-      // Group by current (engine) status:
-      Map<CbasRunStatus, List<Run>> engineStatuses =
-          updatableRuns.stream()
-              .collect(
-                  groupingBy(
-                      r -> {
-                        long getStatusStartNanos = System.nanoTime();
-                        boolean getStatusSuccess = false;
-                        try {
-                          var result =
-                              CbasRunStatus.fromValue(
-                                  cromwellService.runStatus(r.engineId()).getState());
-                          getStatusSuccess = true;
-                          return result;
-                        } catch (ApiException | IllegalArgumentException e) {
-                          logger.warn("Unable to fetch updated status for run {}.", r.runId(), e);
-                          return r.status();
-                        } finally {
-                          recordOutboundApiRequestCompletion(
-                              "wes/runStatus", getStatusStartNanos, getStatusSuccess);
-                        }
-                      }));
-
-      Set<Run> updatedRuns = new HashSet<>(runs);
-
-      for (Map.Entry<CbasRunStatus, List<Run>> engineStateEntry : engineStatuses.entrySet()) {
-        for (Run r : engineStateEntry.getValue()) {
-          updatedRuns.remove(r);
-          updatedRuns.add(updateDatabaseRunStatus(engineStateEntry, r));
+      Set<Run> updatedRuns = new HashSet<>();
+
+      for (Run r : runs) {
```

Comment:
this block is getting picked up as a code smell for "cognitive complexity", but I think it's an improvement over what it's replacing